cql3/statements/describe_statement.cc

```diff
@@ -0,0 +1,811 @@
+        if (with_stmt) { 
+            desc.add_create_statement(get_create_statement(*udf)); 
+        }
+
+        return desc;
+    }));
+}
+
+std::vector<description> aggregate(const sstring& ks, const sstring& name) {
+    auto fs = functions::functions::find(functions::function_name(ks, name));
+    if(fs.empty()) {
+        throw exceptions::invalid_request_exception(format("Aggregate '{}' not found in keyspace '{}'", name, ks));
+    }
+
+    std::vector<description> result;
+    for(auto&& f: fs) {
+        auto uda = dynamic_pointer_cast<functions::user_aggregate>(f.second);
+
+        if (uda) {
+            auto stmt = get_create_statement(*uda);
+            result.push_back(description(f.first.keyspace, "aggregate", f.first.name, stmt));
+        }
+    }
+    return result;
+}
+
+std::vector<description> aggregates(const sstring& ks, bool with_stmt = false) {
+    auto udas = functions::functions::get_user_aggregates(ks);
+    boost::sort(udas, [] (const auto& a, const auto& b) {
+        return a->name().name < b->name().name;
+    });
+
+    return boost::copy_range<std::vector<description>>(udas | boost::adaptors::transformed([&ks, &with_stmt] (auto&& uda) {
+        auto desc = description(uda->name().keyspace, "aggregate", uda->name().name);
+        if (with_stmt) {
+            desc.add_create_statement(get_create_statement(*uda));
+        }
+
+        return desc;
+    }));
+}
+
+description view(const data_dictionary::database& db, const sstring& ks, const sstring& name, bool with_internals) {
+    auto view = db.try_find_table(ks, name);
+    if (!view || !view->schema()->is_view()) {
+        throw exceptions::invalid_request_exception(format("Materialized view '{}' not found in keyspace '{}'", name, ks));
+    }
+
+    std::ostringstream os;
+    view->schema()->describe(db.real_database(), os, with_internals);
+    auto stmt = os.str();
+
+    return description(ks, "view", name, stmt);
+}
+
+description index(const data_dictionary::database& db, const sstring& ks, const sstring& name, bool with_internals) {
+    auto schema = db.find_indexed_table(ks, name);
+    if (!schema) {
+        throw exceptions::invalid_request_exception(format("Table for existing index '{}' not found in '{}'", name, ks));
+    }
+
+    std::ostringstream os;
+    std::optional<view_ptr> idx;
+
+    auto views = db.find_table(ks, schema->cf_name()).views();
+    for (const auto& view: views) {
+        if (is_index(db, view) && secondary_index::index_name_from_table_name(view->cf_name()) == name) {
+            idx = view;
+            break;
+        }
+    }
+
+    (*idx)->describe(db.real_database(), os, with_internals);
+    auto stmt = os.str();
+
+    return description(ks, "index", name, stmt);
+}
+
+std::vector<description> table(const data_dictionary::database& db, const sstring& ks, const sstring& name, bool with_internals) {
+    auto table = db.try_find_table(ks, name);
```

Comment:
This will surely stall with 5000 tables. So need to return a future<std::vector<description>> and convert to a coroutine with maybe_yields.

Reply:
Or rather, it will stall with 5000 indexes (which we'll never have for a single table), but better make it a coroutine as the next function will have to be one.

Reply:
futurized methods that describe more than one element and added maybe_yield()